src/util.rs

```diff
@@ -313,6 +297,44 @@ pub trait ReadExt<'data> {
     fn read_u64(&mut self) -> Option<u64> {
         self.read_pod::<u64>()
     }
+
+    /// Read a `u128` encoded as unsigned variable length little endian base 128
+    /// value.
+    ///
+    /// The function returns the value read along with the number of bytes
+    /// consumed.
+    fn read_u128_leb128(&mut self) -> Option<(u128, u8)> {
+        let mut shift = 0;
+        let mut value = 0u128;
+        while let Some(bytes) = self.read_slice(1) {
```

Comment:
does rust support something like

```
while let Some([bytes]) = self.read_slice(1) {
...
}
```

?

Reply:
Yes, and I had that before, but it's not quite the same semantics: we would exit the loop when the slice has anything but a single element. Right now there is no really a difference, because we know that `read_slice(1)` returns a slice with a single element. But if we, hypothetically, were to adjust `self.read_slice(1)` to read two bytes we'd now silently have a bug in the code, whereas right now we panic instead.